src/libraries/Microsoft.Extensions.Logging.Console/src/JsonConsoleLogFormatter.cs

```diff
@@ -0,0 +1,168 @@
+                    writer.WriteNumber("hResult", exception.HResult);
+                    writer.WriteEndObject();
+                }
+
+                GetScopeInformation(writer, scopeProvider);
+
+                writer.WriteEndObject();
+
+                writer.Flush();
+            }
+            textWriter.Write(Encoding.UTF8.GetString(output.WrittenMemory.Span));
+            textWriter.Write(Environment.NewLine);
+        }
+
+
+        private static string GetLogLevelString(LogLevel logLevel)
+        {
+            switch (logLevel)
+            {
+                case LogLevel.Trace:
+                    return "json_trace";
+                case LogLevel.Debug:
+                    return "json_debug";
+                case LogLevel.Information:
+                    return "json_info";
+                case LogLevel.Warning:
+                    return "json_warn";
+                case LogLevel.Error:
+                    return "json_fail";
+                case LogLevel.Critical:
+                    return "json_critical";
+                default:
+                    throw new ArgumentOutOfRangeException(nameof(logLevel));
+            }
+        }
+
+        private void GetScopeInformation(Utf8JsonWriter writer, IExternalScopeProvider scopeProvider)
+        {
+            try
+            {
+                if (FormatterOptions.IncludeScopes && scopeProvider != null)
+                {
+                    writer.WriteStartObject("scopes");
+                    scopeProvider.ForEachScope((scope, state) =>
+                    {
+                        if (scope is IReadOnlyList<KeyValuePair<string, object>> kvps)
+                        {
+                            foreach (var kvp in kvps)
+                            {
+                                if (kvp.Value is string ss)
+                                    state.WriteString(kvp.Key, ss);
+                                else
+                                if (kvp.Value is int ii)
+                                    state.WriteNumber(kvp.Key, ii);
+                                else
+                                {
+                                    // check how this work
+                                    state.WritePropertyName(kvp.Key);
+                                    JsonSerializer.Serialize(state, kvp.Value);
+                                }
+                            }
+                            //state is the writer
+                            //JsonSerializer.Serialize(state, scope);
+                        }
+                        else
+                        {
+                            state.WriteString("noName", scope.ToString());
+                        }
+                    }, (writer));
+                    writer.WriteEndObject();
+                }
+            }
+            catch (Exception ex)
+            {
+                System.Console.WriteLine("Something went wrong" + ex.Message);
```

Comment:
Debugging code?

Reply:
I am wondering what would be the case when getting exception here? do we'll end up with mal-formatted Json text? 

---
In reply to: [451134311](https://github.com/dotnet/runtime/pull/38616#discussion_r451134311) [](ancestors = 451134311)

Reply:
I removed it and simplified in the GetScopeInformation logic.